src/Compilers/Core/Portable/DiagnosticAnalyzer/AsyncQueue.cs

```diff
@@ -140,6 +146,11 @@ public void Complete()
             }
         }
 
+        public void PromiseNotToEnqueue()
```

Comment:
Could `TryComplete` and `PromiseNotToEnqueue` be replaced with `Complete` and `Cancel` where the `Compiler` calls `Complete` and the `AnalyzerDriver` calls `Cancel`?